### PR TITLE
Trading

### PR DESCRIPTION
**Pull request by JamesEVerhagen**
Adding in trading to the new state machine

### CODE REVIEW

File: src/monopoly/view/monopoly_view.js [hunk 1]
```diff
@@ -362,7 +417,7 @@ MonopolyView.prototype.drawCommunityChestTile = function(x_pos, y_pos, property)
     tile.drawRect(0, 0, constants.tileShortSide, constants.tileLongSide);
 
     // create a texture from an image path
-    var texture = PIXI.Texture.fromImage('assets/community_chest.png');
```
Comment (qstearns) [inline src/monopoly/view/monopoly_view.js]: We should be using the png now

Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: @JamesEVerhagen can you fix this so we can merge?

File: src/monopoly/view/monopoly_view.js [hunk 2]
```diff
@@ -883,71 +939,117 @@ MonopolyView.prototype.drawMessage = function() {
...
+    btn.x = x;
+    btn.y = 180;
+    btn.beginFill(color, 1);
+    btn.drawRect(0, 0, 200, 50);
+    container.addChild(btn);
+    var context = this;
+    btn.interactive = true;
+    btn.click = function(mouseData) {
+        if(money){
+            context.game.trade.money += money;
```
Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: we should be calling methods of the game to update this and not doing the update itself in the view

File: src/monopoly/view/monopoly_view.js [hunk 3]
```diff
@@ -883,71 +939,117 @@ MonopolyView.prototype.drawMessage = function() {
...
+        } else {
+            context.turnMap.updateState(ans);
```
Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: Same as ^

File: src/monopoly/view/monopoly_view.js [hunk 4]
```diff
@@ -963,7 +1065,23 @@ MonopolyView.prototype.updateProperties = function() {
...
         }
     }
-}
+};
+
+MonopolyView.prototype.updateTradeInfo = function(){
+    
+    if(this.turnMap.getCurrentState() == PROPOSE_TRADE){
+        if(!this.game.trade){
+            this.game.createTrade();
```
Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: You should create the trade in the _onEnter in the state machine and not in the view

File: src/monopoly/view/monopoly_view.js [hunk 5]
```diff
@@ -963,7 +1065,23 @@ MonopolyView.prototype.updateProperties = function() {
...
+        }
+        var trade_player_name = "";
+        if(!this.game.trade.answering_player){
```
Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: add a space `if (...) {`

File: src/monopoly/view/monopoly_view.js [hunk 6]
```diff
@@ -963,7 +1065,23 @@ MonopolyView.prototype.updateProperties = function() {
...
+            trade_player_name = "Select Player";
+        }else{
```
Comment (andrewjmeier) [inline src/monopoly/view/monopoly_view.js]: add a space `} else {`

**Comment by andrewjmeier** [comment]
@JamesEVerhagen we need to get this merged in today can you fix this stuff?

File: src/monopoly/view/monopoly_view.js [hunk 7]
```diff
@@ -242,15 +241,69 @@ MonopolyView.prototype.drawCard = function(xPos, yPos, width, height, text, colo
...
+    box.lineStyle(1, 0, 1);
+    box.beginFill(0x44C0DF, 0);
+    box.x = x_pos;
+    box.y = y_pos;
+    box.pivot.set(constants.tileShortSide, constants.tileLongSide);
+    box.drawRect(0, 0, constants.tileShortSide, constants.tileLongSide);
+
+    this.setupPropertyClick(box, property);
+    
+    box.rotation = rotation;
```
Comment (konnorbeard) [inline src/monopoly/view/monopoly_view.js]: where's rotation set here?

File: src/monopoly.js [hunk 8]
```diff
@@ -27,7 +27,9 @@ var monopoly = new Game(players, board, turn);
 
 var turn = new Turn(monopoly);
 
-var view = new MonopolyView(monopoly, turn);
+monopoly.setTurn(turn);
```
Comment (JamesEVerhagen) [inline src/monopoly.js]: Set turnMap of the game after we create the turn. If I don't do this then the game can find the turnMap. Not sure if there is a better way of doing this.

File: src/monopoly/view/monopoly_view.js (hunk 7 above)
Comment (JamesEVerhagen) [inline src/monopoly/view/monopoly_view.js]: just changed this to make more intuitive

**Comment by andrewjmeier** [comment]
Fix your merge conflicts and then go for it :+1:

**Pull request merged**